auparse/auparse.go

```diff
@@ -514,78 +533,78 @@ func execveArgs(data map[string]*field) error {
 
 // parseSELinuxContext parses a SELinux security context of the form
 // 'user:role:domain:level:category'.
-func parseSELinuxContext(key string, data map[string]*field) error {
-	field, found := data[key]
-	if !found {
-		return fmt.Errorf("%v key not found", key)
+func (fm fieldMap) parseSELinuxContext(key string) error {
+	f, err := fm.find(key)
+	if err != nil {
+		return err
 	}
 
 	keys := []string{"_user", "_role", "_domain", "_level", "_category"}
-	contextParts := strings.SplitN(field.Value(), ":", len(keys))
+	contextParts := strings.SplitN(f.value, ":", len(keys))
 	if len(contextParts) == 0 {
 		return fmt.Errorf("failed to split SELinux context field %v", key)
 	}
-	delete(data, key)
+	fm.delete(key)
 
 	for i, part := range contextParts {
-		data[key+keys[i]] = newField(part)
+		fm.add(key+keys[i], newField(part))
 	}
 	return nil
 }
 
-func result(data map[string]*field) error {
+func (fm fieldMap) result() error {
 	// Syscall messages use "success". Other messages use "res".
-	field, found := data["success"]
-	if !found {
-		field, found = data["res"]
-		if !found {
+	field, err := fm.find("success")
+	if err != nil {
+		field, err = fm.find("res")
+		if err != nil {
 			return errors.New("success and res key not found")
```

Comment:
Make this a package var.

Reply:
I didn't change this specific error in this PR, only the logic around it, but yeah it makes sense to add a var